app/controllers/packages_controller.rb

```diff
@@ -0,0 +1,73 @@
+require 'httparty'
+
+class PackagesController < ApplicationController
+ # include Timeout
+
+  def rates
+    origin = get_origin
+    package = get_package
+    destination = get_destination
+
+     ups = ActiveShipping::UPS.new(login: ENV["UPS_LOGIN"], password: ENV["UPS_PASSWORD"], key: ENV["UPS_KEY"])
+     usps = ActiveShipping::USPS.new(login: ENV["USPS_LOGIN"])
+
+    ups_response = ups.find_rates(origin, destination, package)
+    ups_rates = ups_response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price, rate.delivery_date]}
+
+    usps_response = usps.find_rates(origin, destination, package)
+    usps_rates = usps_response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price, rate.delivery_date]}
+
```

Comment:
Lines 14-15 & 17-18 seem like copy pasta code (i.e. identical). Could this have been pulled out into its own method?

````
def collectRates(shipper, origin, destination package)
    # returns sorted rates
end
````